### ISSUE TITLE
MS income tax incorrectly allows unlimited capital losses to offset income

### ISSUE THREAD

**Issue by PavelMakarchuk** [PolicyEngine/policyengine-us#6964]
## Summary

Mississippi income tax calculation allows unlimited capital losses to offset ordinary income, instead of using the federal $3,000 capital loss limit. This results in $0 tax when there should be ~$17,136 tax.

## Root Cause

`policyengine_us/parameters/gov/states/ms/tax/income/income_sources.yaml` includes `capital_gains` which is the raw sum of `short_term_capital_gains + long_term_capital_gains` without applying the federal capital loss limitation.

MS Form 80-105 Line 40 ("Capital gain (loss)") flows from federal Schedule D, which already has the $3,000 loss limit applied. PE should use the loss-limited amount.

## Test Case

- Filing: MS joint 2024
- Interest income: $404,179
- Short-term capital gains: -$9,670,468
- Long-term capital gains: +$4,900,962
- Net capital loss: -$4,769,506 (limited to -$3,000 federally)

**Expected (per TaxAct Form 80-105):**
- MS AGI per spouse: ~$200,589
- MS Tax: $17,136

**Actual (PE):**
- MS AGI: $0 (because unlimited -$4.77M loss offsets $404K interest)
- MS Tax: $0

## Files to Fix

1. Create new person-level variable `loss_limited_capital_gains` that allocates `loss_limited_net_capital_gains` (TaxUnit) proportionally to persons
2. Update `policyengine_us/parameters/gov/states/ms/tax/income/income_sources.yaml` to use `loss_limited_capital_gains` instead of `capital_gains`

## Suggested Implementation

```python
# New file: policyengine_us/variables/household/income/person/capital_gains/loss_limited_capital_gains.py
from policyengine_us.model_api import *

class loss_limited_capital_gains(Variable):
    value_type = float
    entity = Person
    label = "Loss-limited capital gains (person's share)"
    unit = USD
    definition_period = YEAR

    def formula(person, period, parameters):
        # Get person's raw capital gains
        person_capital_gains = person("capital_gains", period)
        
        # Get tax unit totals
        tax_unit_capital_gains = person.tax_unit.sum(person_capital_gains)
        tax_unit_loss_limited = person.tax_unit("loss_limited_net_capital_gains", period)
        
        # If no capital gains, return 0
        # If gains (positive), return person's share unchanged
        # If losses, allocate the loss-limited amount proportionally
        is_loss = tax_unit_capital_gains < 0
        
        # For losses: allocate loss_limited amount proportionally
        proportion = where(
            tax_unit_capital_gains != 0,
            person_capital_gains / tax_unit_capital_gains,
            0
        )
        
        return where(
            is_loss,
            tax_unit_loss_limited * proportion,
            person_capital_gains
        )
```

Then update `income_sources.yaml`:
```yaml
# Change capital_gains to loss_limited_capital_gains
- loss_limited_capital_gains  # Was: capital_gains
```

## Integration Test

```yaml
- name: MS joint with capital losses should use federal $3,000 limit
  absolute_error_margin: 100
  period: 2024
  input:
    people:
      person1:
        age: 40
        short_term_capital_gains: -4835234
        long_term_capital_gains: 2450480.75
        taxable_interest_income: 202089.485
        is_tax_unit_head: true
      person2:
        age: 40
        short_term_capital_gains: -4835234
        long_term_capital_gains: 2450480.75
        taxable_interest_income: 202089.485
        is_tax_unit_spouse: true
    tax_units:
      tax_unit:
        members: [person1, person2]
    spm_units:
      spm_unit:
        members: [person1, person2]
    households:
      household:
        members: [person1, person2]
        state_fips: 28
  output:
    ms_income_tax: 17136  # From TaxAct Form 80-105
```

## References

- [MS Form 80-105 Instructions](https://www.dor.ms.gov/sites/default/files/Forms/Individual/80100241.pdf) - Line 40: "Capital gain (loss) (attach Federal Schedule D)"
- [Federal Schedule D](https://www.irs.gov/pub/irs-pdf/f1040sd.pdf) - Line 21 limits capital losses to $3,000
- Related TAXSIM issue: https://github.com/PolicyEngine/policyengine-taxsim/issues/617

## Note

This fix may affect other states that use `capital_gains` in their income sources. Review needed for: any state that starts from federal AGI or uses federal Schedule D amounts should use `loss_limited_capital_gains` instead of `capital_gains`.

## Related Issues

- PolicyEngine/policyengine-taxsim#617: MS 2024 joint 4900Kltcg -9670Kstcg

**Comment by PavelMakarchuk** [comment]
@PolicyEngine review the issue and implement the suggested fix.
